models/MARBL_joint_estimation/model_mod.f90

```diff
@@ -0,0 +1,654 @@
+
+! Read values from the namelist
+
+call find_namelist_in_file("input.nml", "model_nml", iunit)
+read(iunit, nml = model_nml, iostat = io)
+
+call check_namelist_read(iunit, io, "model_nml")
+
+! Record the namelist values used for the run 
+if (do_nml_file()) write(nmlfileunit, nml=model_nml)
+if (do_nml_term()) write(     *     , nml=model_nml)
+
+! This time is both the minimum time you can ask the model to advance
+! (for models that can be advanced by filter) and it sets the assimilation
+! window.  All observations within +/- 1/2 this interval from the current
+! model time will be assimilated. If this is not settable at runtime 
+! feel free to hardcode it and remove from the namelist.
+assimilation_time_step = set_time(time_step_seconds, &
+                                  time_step_days)
+
+! setting up the DART state vector
+call verify_state_variables(model_state_variables, nfields, variable_table, &
+                            state_qty_list, state_clamp_vals, update_var_list)
+
+    state_dom_id = add_domain(template_file(1), nfields, &
+                              var_names = variable_table(1:nfields, 1), &
+                              kind_list = state_qty_list(1:nfields), &
+                              clamp_vals = state_clamp_vals, &
+                              update_list = update_var_list(1:nfields))
+
+! setting up the DART parameter vector
+if(estimate_params) then
+    call verify_state_variables(model_parameters, nfields, param_table, &
+                                param_qty_list, param_clamp_vals, update_param_list)
+
+    param_dom_id = add_domain(template_file(2), nfields, &
+                              var_names = param_table(1:nfields, 1), &
+                              kind_list = param_qty_list(1:nfields), &
+                              clamp_vals = param_clamp_vals, &
+                              update_list = update_param_list(1:nfields))
+end if
+
+call read_num_layers     ! setting the value of nz
+call read_ocean_geometry ! determining the basin depth
+
+end subroutine static_init_model
+
+!------------------------------------------------------------------
+! Reads the simulation length from a netCDF file.
+
+function read_model_time(filename)
+
+character(len=*), intent(in) :: filename
+type(time_type) :: read_model_time
+
+integer :: ncid
+character(len=*), parameter :: routine = 'read_model_time'
+real(r8) :: days
+type(time_type) :: mom6_time
+integer :: mom_base_date_in_days, mom_days
+
+mom_base_date_in_days = 139157 ! 1982 1 1 0 0
+ncid = nc_open_file_readonly(filename, routine)
+
+call nc_get_variable(ncid, 'Time', days, routine)
+
+call nc_close_file(ncid, routine)
+    
+mom_days = int(days) + mom_base_date_in_days
+
+read_model_time = set_time(0,mom_days)
+
+end function read_model_time
+
+!------------------------------------------------------------------
+! Returns the number of items in the state vector as an integer. 
+
+function get_model_size()
```

Comment:
This is not very clean, especially if the parameter domain is not active. I would count the domains and loop over them to get the `model_size` (which is by the way defined on line 84 but not used). I suggest a code similar to the one below: 
```
domain_count = get_num_domains()
model_size = 0
do i_dom = 1,domain_count
     model_size = model_size + get_domain_size(i_dom)
enddo
```
This is related to my previous comment on active domains and the nature of the DA configuration